### ISSUE TITLE
Reorganize WallEntry context menu

### ISSUE THREAD

**Issue by buddh4** [humhub/humhub#3446]
### Is your feature request related to a problem? Please describe.

Currently the WallEntry context menu is kind of a mess e.g. in case of the polls module:

![grafik](https://user-images.githubusercontent.com/9800854/52164066-beb1bc80-26eb-11e9-8d09-c2118ea24efd.png)

#### Seperate the options into groups seperated by a divider:

**Edit Section**

Contains main edit options (often used)

- Delete
- Edit
- Topics

**Content section**

Content specific options (+permalink)

- Permalink
- Close Poll
- Reopen Poll
- Reset Answers

> Note: Content specific options should rather be included into the wallentry output itself.

**More section**

More section for features not that commonly used and hidden by a collapsable sub menu item "... more"

- Turn on/off notification
- Change visibility (public/private)
- Pin to top
- Move content
- Move to archive
- Print
- Content Infos
- ....

#### Add submenus 

Modules should be able to add submenus

### Question:

- We need to define an actual order of default items
- We need to define a sortOrder range for the given sections (e.g. Entries with sortOrder > 1000 are added to the more section etc.

**Comment by Felli** [comment]
Here's an idea for the polls module, why not just make it a button next to the upload/submit button, and then add the Open/Close Poll & Reset Answers within the post itself? Would clear up some of the clutter while keeping the module in a good view.

**Comment by buddh4** [comment]
Yeah, forgot to add the solution to rather include such things in the wallentry output itself. Thanks.

**Comment by luke-** [comment]
Partly in v1.4 - Need more work in v1.5

**Comment by buddh4** [comment]
Just as an update in 1.4 the order and seperation is as follows:

![grafik](https://user-images.githubusercontent.com/9800854/66665993-f7793c80-ec4f-11e9-8636-de85ab41838f.png)

In case someone has a better idea/concept we can align the order and seperation in beta.
Also a *show more* would be great in my opinion.

**Comment by Ace2213** [comment]
Topics should be lower, near "Pin to top", since both are for organizing posts. Right below them would be Move content and Move to archive. Those last 4 options can be separated as a management/admin section

Edit should be first imo to make it harder to accidentally click it, then Delete then Permalink. Next section privacy switcher and notifications. Final section is for admin options. Topics, Pin to top, Move content and finally Move to archive.

Content-specific options like reset poll can be either at the top or in the wall entry itself as a button. Having them in the wall entry will make them more discoverable, but the poll wall entry already has a lot of stuff going on, if you enable all the options.

**Comment by luke-** [comment]
Sounds very good!

I would prefer a collapsible section for less commonly used links. 
That could be, for example:
- Move
- Change visibility
- Delete?
- Pin
- Archive

**Comment by MasterMindNET** [comment]
> ### Question:
> * We need to define an actual order of default items
> * We need to define a sortOrder range for the given sections (e.g. Entries with sortOrder > 1000 are added to the more section etc.

Do you mean the sortOrder must be greater than 1000, right?
And how would you like to see the sortOrder editing? somewhere in the settings or where? or just in code?

**Comment by Carl2213** [comment]
I think a collapsible section would add more complexity. Personally, I would disable it.

Is the permalink accessible even if the button is disabled? Because I think a share button would completely negate its necessity. See [here](https://community.humhub.com/content/perma?id=224553).

As for sectioning/order, here's my suggestion:
![image](https://user-images.githubusercontent.com/69092633/93690321-51fc5900-faad-11ea-9de5-48b0371846e8.png)


The last section is for organizing content, which is often only possible by admins. So non-admins would only see the first two sections.

Those without permissions would also have a pretty clean context menu:
![image](https://user-images.githubusercontent.com/69092633/93690313-414be300-faad-11ea-95ad-03745b8240b3.png)

**Comment by MasterMindNET** [comment]
1. > I think a collapsible section would add more complexity. Personally, I would disable it.

So do you propose just separate the link for categories by some line (like there is on your screenshots, right?)
As I can see the options(delete and edit only available for not if you do not have rights)

2. >See here.

I have answered your questions on the [HumHub page](https://community.humhub.com/s/module-development/?contentId=224553), please take a look at my answers.

3. >Is the permalink accessible even if the button is disabled? 

What do you mean by this? if there is a check on the backend if you would try to enter the link to the URL bar or what?

4. >The last section is for organizing content, which is often only possible by admins. So non-admins would only see the first two sections.

For now, we do not have the functionality to edit the order right(only in code)? 
Would you like to add functionality for admins to edit these orders? if yes, where these settings should be, and as far as I am concerned this link order would be the same throughout the whole website.

**Comment by Carl2213** [comment]
> For now, we do not have the functionality to edit the order right(only in code)?
> Would you like to add functionality for admins to edit these orders? if yes, where these settings should be, and as far as I am concerned this link order would be the same throughout the whole website.

This issue is about changing the default order of the context menu. I believe it would still be possible for super(global) admins to edit it through the code (not sure where). I don't think it's necessary to add the ability to customize the order via UI.

> Is the permalink accessible even if the button is disabled?

The permalink functionality is to share a direct link to the post. The socialshare module adds dedicated buttons to each post that allow you to share the permalink to certain social networks. Adding support for WebShare API will replace all these buttons (on supported devices) with a single "external share" button, that will pull up your device's native share menu so you can share to any app you want or copy to clipboard. Thus eliminating the need for the "permalink" context menu entry.

The reason I suggest this is because "Permalink" is the most difficult entry to organize. It doesn't "fit" in any section.

> So do you propose just separate the link for categories by some line (like there is on your screenshots, right?)
> As I can see the options(delete and edit only available for not if you do not have rights)

Yes. The first category/section will only be visible to those that have delete and edit rights (usually the author or admins).  The last category will only be visible to admins. Those with rights to organize posts.

Now that you mention it, Topics can sometimes be manageable by non-admins. So maybe this layout would be better:
Full rights:
![image](https://user-images.githubusercontent.com/69092633/93722584-49b82280-fb6e-11ea-8912-0a27a7954ccd.png)

Non-admin author in space with topics permission:
![image](https://user-images.githubusercontent.com/69092633/93722621-90a61800-fb6e-11ea-93e0-9061eb2b1dd1.png)

No rights, but logged in:
![image](https://user-images.githubusercontent.com/69092633/93722416-f1ccec00-fb6c-11ea-9caf-543ba4ec8800.png)


Those without any rights will only see the turn on notification button.

**Comment by Carl2213** [comment]
If you want to keep the permalink entry, you can do sections of 3:

![image](https://user-images.githubusercontent.com/69092633/93722725-46716680-fb6f-11ea-98ab-5319b39fac17.png)
